gamemodes/terrortown/entities/entities/ttt_smoke.lua

```diff
@@ -0,0 +1,428 @@
+---
+-- smoke handler that does psychic damage
+-- @class ENT
+-- @section ttt_smoke
+
+AddCSLuaFile()
+
+ENT.Type = "anim"
+ENT.Base = "base_anim"
+-- ENT.Model = Model("models/hunter/misc/sphere1x1.mdl")
+ENT.Model = Model("models/weapons/w_eq_smokegrenade_thrown.mdl")
+
+ENT.RenderGroup = RENDERGROUP_TRANSLUCENT
+ENT.SmokeEmissionSound = Sound("npc/env_headcrabcanister/hiss.wav")
+ENT.LoopSound = nil --[[@as CSoundPatch]]
+ENT.Emitter = nil --[[@as CLuaEmitter]]
+ENT.SmokeParticles = {
+	Model("particle/particle_smokegrenade"),
+	Model("particle/particle_noisesphere")
+}
+ENT.SmokeColors = {
+	black       = Color(  0,   0,   0, 255),
+	blue        = Color(  0,   0, 255, 255),
+	cyan        = Color( 14, 236, 232, 255),
+	darkblue    = Color(  0,   0, 100, 255),
+	darkgreen   = Color(  0, 100,   0, 255),
+	darkpurple  = Color( 51,  16,  75, 255),
+	green       = Color(  0, 255,   0, 255),
+	lightgray   = Color(200, 200, 200, 255),
+	lightpurple = Color(140,  23, 217, 255),
+	lime        = Color(175, 237,  18, 255),
+	olive       = Color(100, 100,   0, 255),
+	orange      = Color(250, 100,   0, 255),
+	pink        = Color(255,   0, 255, 255),
+	purple      = Color(100,  18, 155, 255),
+	red         = Color(255,   0,   0, 255),
+	white       = Color(255, 255, 255, 255),
+	yellow      = Color(200, 200,   0, 255),
+}
+ENT.DriftVelocity = 1.5
+ENT.DispersalVelocity = 5
+ENT.RollVariance = 180
+ENT.RollDeltaVariance = 0.1
+ENT.AirResistance = 600
+ENT.DisperseAirResistMultiplier = 0.5
+ENT.DisperseGravityMultiplier = 0.25
+ENT.Bounce = 0.4
+ENT.ScaleFactor = 25
+ENT.Particles = {}
+ENT.ParticleData = {}
+ENT.DebugViz = false
+ENT.DebugRange = true
+
+---
+-- @realm shared
+function ENT:SetupDataTables()
+	self:NetworkVar("String", 0, "SharedSeed")
+
+	self:NetworkVar("Float", 0, "LifeTime")
+	self:NetworkVar("Float", 1, "FadeTime")
+
+	self:NetworkVar("Bool", 0, "RandomColor")
+	self:NetworkVar("Bool", 1, "ExtinguishFires")
+
+	self:NetworkVar("Int", 0, "Density")
+	self:NetworkVar("Int", 1, "Radius")
+
+	self:NetworkVar("Int", 2, "ColorBase")
+	self:NetworkVar("Int", 3, "ColorVariance")
+	self:NetworkVar("Int", 4, "StartSizeBase")
+	self:NetworkVar("Int", 5, "StartSizeVariance")
+	self:NetworkVar("Int", 6, "EndSizeBase")
+	self:NetworkVar("Int", 7, "EndSizeVariance")
+
+	-- extra, non-smoke-data-reflecting
+	self:NetworkVar("Vector", 0, "SmokeColor")
+	self:NetworkVar("Float", 2, "CreateTime")
+	self:NetworkVar("Bool", 2, "Alive")
+	self:NetworkVar("Float", 3, "DeactivateTime")
+end
+
+---
+-- @realm shared
+function ENT:Initialize()
+	if self.DebugViz then
+		self:SetMaterial("models/shiny.vtf")
+		self:SetColor( Color(255, 200, 0, 128) )
+		self:SetRenderMode( RENDERMODE_TRANSCOLOR )
+		-- self:SetNoDraw(false)
+		-- self:DrawShadow(true)
+	else
+		self.RenderGroup = RENDERGROUP_TRANSLUCENT
+		self:DrawShadow(false)
+		self:SetNoDraw(false)
+		self:SetRenderMode( RENDERMODE_TRANSCOLOR )
+	end
+
+	--[[ 4: continue as normal ]]--
```

Comment:
```suggestion
	-- 4: continue as normal
```

but where are 1 to 3?